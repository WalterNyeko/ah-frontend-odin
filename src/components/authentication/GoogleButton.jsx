import React from 'react';
import PropTypes from 'prop-types';
import GoogleLogin from 'react-google-login';
import GoogleIcon from 'components/icons/Google';

const GoogleButton = ({ onSuccessHandler }) => (
  <GoogleLogin
    clientId="95888263107-0i151ro1iaqbdlecpssbdbte1bno6kfs.apps.googleusercontent.com"
    buttonText="Login"
    onSuccess={onSuccessHandler}
    render={renderProps => (
      <button onClick={renderProps.onClick} type="button" className="btn btn-link googleIcon">
        {' '}
        <GoogleIcon />
      </button>
    )}
  />
);

GoogleButton.propTypes = {
  onSuccessHandler: PropTypes.func.isRequired,
};

export default GoogleButton;
