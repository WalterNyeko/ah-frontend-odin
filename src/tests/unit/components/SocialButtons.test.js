import React from 'react';
import renderer from 'react-test-renderer';
import FacebookButton from 'components/authentication/FaceBookButton';
import GoogleButton from 'components/authentication/GoogleButton';

describe('Test social login buttons', () => {
  test('Google Button', () => {
    const wrapper = renderer.create(<GoogleButton onSuccessHandler={jest.fn()} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Facebook Button', () => {
    const wrapper = renderer.create(<FacebookButton onSuccessHandler={jest.fn()} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
