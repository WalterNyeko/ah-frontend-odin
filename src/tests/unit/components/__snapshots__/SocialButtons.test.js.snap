// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test social login buttons Facebook Button 1`] = `
<button
  className="btn btn-link facebookIcon"
  onClick={[Function]}
  type="button"
>
   
  <svg
    viewBox="0 0 264 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M76.7 512V283H0v-91h76.7v-71.7C76.7 42.4 124.3 0 193.8 0c33.3 0 61.9 2.5 70.2 3.6V85h-48.2c-37.8 0-45.1 18-45.1 44.3V192H256l-11.7 91h-73.6v229"
    />
  </svg>
</button>
`;

exports[`Test social login buttons Google Button 1`] = `
<button
  className="btn btn-link googleIcon"
  onClick={[Function]}
  type="button"
>
   
  <svg
    viewBox="0 0 488 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
    />
  </svg>
</button>
`;
